core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -255,7 +260,7 @@ ManifestGroup planWith(ExecutorService newExecutorService) {
                 entry -> entry.status() != ManifestEntry.Status.EXISTING);
           }
 
-          if (fileFilter != null && fileFilter != Expressions.alwaysTrue()) {
+          if (evaluator != null) {
```

Comment:
Would it work if we simply define the evaluator only needed, like below?

```
          if (fileFilter != null && fileFilter != Expressions.alwaysTrue()) {
            Evaluator evaluator = new Evaluator(DataFile.getType(EMPTY_STRUCT), fileFilter, caseSensitive);
            entries = CloseableIterable.filter(entries,
                entry -> evaluator.eval((GenericDataFile) entry.file()));
          }
```

Reply:
We instantiate it above so that we only create one and share it for all of the evaluations. It just cuts down on Evaluator creation and garbage collection.